### ISSUE TITLE
errors when using npm run serve: typescript and vuetify not compatible?

### ISSUE THREAD

**Issue by WilfriedVanDyk** [WilfriedVanDyk/invoerApp1#1]
following errors are showing when serving project: see below
(also not able to build project).
Tried to solve problem with using solution in "https://github.com/vuetifyjs/vuetify/issues/11683"
did not work...
thanks for helping out!

ERROR in C:/Users/Wilfried/source/invoerAppVuetify/invoerapp/src/main.ts(12,3):
12:3 No overload matches this call.
  Overload 1 of 3, '(options?: ThisTypedComponentOptionsWithArrayProps<Vue, object, object, object, never> | undefined): CombinedVueInstance<Vue, object, object, object, Record<...>>', gave the following error.
    Argument of type '{ router: VueRouter; store: Store<{}>; vuetify: any; render: (h: CreateElement) => VNode; }' is not assignable to parameter of type 'ThisTypedComponentOptionsWithArrayProps<Vue, object, object, object, never>'.
      Object literal may only specify known properties, and 'vuetify' does not exist in type 'ThisTypedComponentOptionsWithArrayProps<Vue, object, object, object, never>'.   
  Overload 2 of 3, '(options?: ThisTypedComponentOptionsWithRecordProps<Vue, object, object, object, object> | undefined): CombinedVueInstance<Vue, object, object, object, Record<...>>', gave the following error.
    Argument of type '{ router: VueRouter; store: Store<{}>; vuetify: any; render: (h: CreateElement) => VNode; }' is not assignable to parameter of type 'ThisTypedComponentOptionsWithRecordProps<Vue, object, object, object, object>'.
      Object literal may only specify known properties, and 'vuetify' does not exist in type 'ThisTypedComponentOptionsWithRecordProps<Vue, object, object, object, object>'. 
  Overload 3 of 3, '(options?: ComponentOptions<Vue, DefaultData<Vue>, DefaultMethods<Vue>, DefaultComputed, PropsDefinition<Record<string, any>>, Record<...>> | undefined): 
CombinedVueInstance<...>', gave the following error.
    Argument of type '{ router: VueRouter; store: Store<{}>; vuetify: any; render: (h: CreateElement) => VNode; }' is not assignable to parameter of type 'ComponentOptions<Vue, DefaultData<Vue>, DefaultMethods<Vue>, DefaultComputed, PropsDefinition<Record<string, any>>, Record<...>>'.
      Object literal may only specify known properties, and 'vuetify' does not exist in type 'ComponentOptions<Vue, DefaultData<Vue>, DefaultMethods<Vue>, DefaultComputed, PropsDefinition<Record<string, any>>, Record<...>>'.
    10 |   router,
    11 |   store,
  > 12 |   vuetify,
       |   ^
    13 |   render: h => h(App)
    14 | }).$mount('#app')
    15 | 
ERROR in C:/Users/Wilfried/source/invoerAppVuetify/invoerapp/src/plugins/vuetify.ts(3,21):
3:21 Could not find a declaration file for module 'vuetify/lib'. 'C:/Users/Wilfried/source/invoerAppVuetify/invoerapp/node_modules/vuetify/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/vuetify` if it exists or add a new declaration (.d.ts) file containing `declare module 'vuetify/lib';`
    1 | import 'material-design-icons-iconfont/dist/material-design-icons.css';
  > 3 | import Vuetify from 'vuetify/lib';
      |                     ^
    4 | import colors from 'vuetify/lib/util/colors.js';
    5 | 
    6 | Vue.use(Vuetify);
ERROR in C:/Users/Wilfried/source/invoerAppVuetify/invoerapp/src/plugins/vuetify.ts(4,20):
4:20 Could not find a declaration file for module 'vuetify/lib/util/colors.js'. 'C:/Users/Wilfried/source/invoerAppVuetify/invoerapp/node_modules/vuetify/lib/util/colors.js' 
implicitly has an 'any' type.
  Try `npm install @types/vuetify` if it exists or add a new declaration (.d.ts) file containing `declare module 'vuetify/lib/util/colors.js';`
    2 | import Vue from 'vue';
    3 | import Vuetify from 'vuetify/lib';
  > 4 | import colors from 'vuetify/lib/util/colors.js';
      |                    ^
    5 | 
    6 | Vue.use(Vuetify);
    7 | 
Version: typescript 3.9.6

**Comment by SightStudio** [comment]
same in here

**Comment by SightStudio** [comment]
i've solved this issue by add vuetify types in tsconfig.json

example 

**tsconfig.json**
```js
"compilerOptions": {
     ...

    "types": [ "vuetify" ],

    ...
}
```

**Comment by WilfriedVanDyk** [comment]
thanks!
this resolved the issue of vuetify/lib!
 not the one with vuetify/lib/util/colors.
Just found a solution at : https://medium.com/@JonUK/creating-a-mobile-web-app-with-vue-vuetify-typescript-dc69bed4cd2d

"After installing Vuetify, I needed to make a tweak to src/shims.vue.d to change how Vuetify types are referenced. This glitch might be resolved by the time you read this but if you face the same issue, I resolved it by adding the following code at the bottom of src/shims.vue.d."

declare module 'vuetify/lib' {
  import 'vuetify/types/lib'
}
and  I also added for colors:
declare module 'vuetify/lib/util/colors.js' {
  import 'vuetify/types/services/colors'
}

builds fine now!

**Comment by WilfriedVanDyk** [comment]
thanks!
this resolved the issue of vuetify/lib!😊

this also helped me finding a solution for vuetify/lib/util/colors.
at :
https://medium.com/@JonUK/creating-a-mobile-web-app-with-vue-vuetify-typescript-dc69bed4cd2d

"After installing Vuetify, I needed to make a tweak to src/shims.vue.d to
change how Vuetify types are referenced. This glitch might be resolved by
the time you read this but if you face the same issue, I resolved it by
adding the following code at the bottom of src/shims.vue.d."

declare module 'vuetify/lib' {
import 'vuetify/types/lib'
}
and I also added for colors:
declare module 'vuetify/lib/util/colors.js' {
import 'vuetify/types/services/colors'
}

builds fine now!
(just some warnings now...)🤔thanks Wilfried


Op ma 20 jul. 2020 om 10:16 schreef Dong Min Seol <notifications@github.com
